cogboard-webapp/src/components/BoardCard/index.js

```diff
@@ -24,6 +28,51 @@ const BoardCard = ({ boardData, className }) => {
   const dispatch = useDispatch();
   const isAdmin = useSelector(({app}) => app.isAdmin);
 
+  const ref = useRef(null);
+  const isLoggedIn = useSelector(({ app }) => !!app.jwToken);
+  const [{ isDragging }, drag] = useDrag({
+    item: { type: ItemTypes.BOARD, id, index },
+    canDrag: isLoggedIn,
+    collect: monitor => ({
+      isDragging: monitor.isDragging()
+    })
+  });
+  const [{ isOver }, drop] = useDrop({
+    accept: ItemTypes.BOARD,
+    hover(item, monitor) {
+      if (!ref.current) {
+        return;
+      }
+
+      const { id: sourceId, index: sourceIndex } = item;
+      const targetIndex = index;
+
+      if (sourceIndex === targetIndex) {
+        return;
+      }
+
+      const { top, bottom } = ref.current.getBoundingClientRect();
+      const dropTargetMiddleY = bottom - (bottom - top) / 2;
+      const { y: dragSourceMouseY } = monitor.getClientOffset();
+
+      if (
+        (sourceIndex < targetIndex && dragSourceMouseY < dropTargetMiddleY) ||
```

Comment:
Can this complicated  `if` logic be named somehow and presented as a method ?

Reply:
done ;)